pkg/prometheus/promcfg.go

```diff
@@ -431,170 +438,93 @@ func (cg *ConfigGenerator) Generate(
 	additionalAlertManagerConfigs []byte,
 	ruleConfigMapNames []string,
 ) ([]byte, error) {
+	cpf := cg.prom.GetCommonPrometheusFields()
+
 	// validates the value of scrapeTimeout based on scrapeInterval
-	if p.Spec.ScrapeTimeout != "" {
-		if err := compareScrapeTimeoutToScrapeInterval(p.Spec.ScrapeTimeout, p.Spec.ScrapeInterval); err != nil {
+	if cpf.ScrapeTimeout != "" {
+		if err := compareScrapeTimeoutToScrapeInterval(cpf.ScrapeTimeout, cpf.ScrapeInterval); err != nil {
 			return nil, err
 		}
 	}
 
+	// Global config
 	cfg := yaml.MapSlice{}
-
-	globalItems := yaml.MapSlice{
-		{Key: "evaluation_interval", Value: p.Spec.EvaluationInterval},
-		{Key: "scrape_interval", Value: p.Spec.ScrapeInterval},
-		{Key: "external_labels", Value: buildExternalLabels(p)},
-	}
-
-	if p.Spec.ScrapeTimeout != "" {
-		globalItems = append(globalItems, yaml.MapItem{
-			Key: "scrape_timeout", Value: p.Spec.ScrapeTimeout,
-		})
-	}
-
-	if p.Spec.QueryLogFile != "" {
-		globalItems = cg.WithMinimumVersion("2.16.0").AppendMapItem(globalItems, "query_log_file", queryLogFilePath(p))
-	}
-
+	globalItems := yaml.MapSlice{}
+	globalItems = cg.AppendEvaluationInterval(globalItems, evaluationInterval)
+	globalItems = cg.AppendScrapeIntervals(globalItems)
+	globalItems = cg.AppendExternalLabels(globalItems)
+	globalItems = cg.AppendQueryLogFile(globalItems, queryLogFile)
```

Comment:
nit: since these methods will only be used inside of this package we could not export them, right?
```suggestion
	globalItems = cg.appendEvaluationInterval(globalItems, evaluationInterval)
	globalItems = cg.appendScrapeIntervals(globalItems)
	globalItems = cg.appendExternalLabels(globalItems)
	globalItems = cg.appendQueryLogFile(globalItems, queryLogFile)
```